tests/src/test/java/alluxio/client/cli/job/DistributedCmdTest.java

```diff
@@ -0,0 +1,145 @@
+  private static final String LOAD = "/load";
+  private static final String MIGRATE = "/migrate";
+  private static final String PERSIST = "/persist";
+  private static final String NOOP_PLAN = "/compact";
+
+  private static final long LOAD_FILE_LENGTH = 1;
+  private static final long MIGRATE_FILE_LENGTH = 2;
+  private static final long PERSIST_FILE_LENGTH = 3;
+  private static final long NOOP_PLAN_FILE_LENGTH = 4;
+  private static final long DEFAULT_ZERO_LENGTH = 0;
+
+  private LoadConfig mLoadJobConfig;
+  private MigrateConfig mMigrateJobConfig;
+  private PersistConfig mPersistJobConfig;
+  private NoopPlanConfig mNoopPlanConfig;
+  private FileSystem mFileSystem;
+  private RetryPolicy mRetryPolicy;
+
+  private URIStatus mLoadURIStatus;
+  private URIStatus mMigrateURIStatus;
+  private URIStatus mPersistURIStatus;
+  private URIStatus mNoopPlanURIStatus;
+
+  @Rule
+  public ExpectedException mException = ExpectedException.none();
+
+  @Before
+  public void before() {
+    mLoadJobConfig = Mockito.mock(LoadConfig.class);
+    mMigrateJobConfig = Mockito.mock(MigrateConfig.class);
+    mPersistJobConfig = Mockito.mock(PersistConfig.class);
+    mNoopPlanConfig = Mockito.mock(NoopPlanConfig.class);
+    mFileSystem = Mockito.mock(FileSystem.class);
+    mRetryPolicy = new CountingRetry(5);
+
+    mLoadURIStatus = Mockito.mock(URIStatus.class);
+    mMigrateURIStatus = Mockito.mock(URIStatus.class);
+    mPersistURIStatus = Mockito.mock(URIStatus.class);
+    mNoopPlanURIStatus = Mockito.mock(URIStatus.class);
+  }
+
+  private void initialize() throws Exception {
+    Mockito.when(mLoadURIStatus.getLength()).thenReturn(LOAD_FILE_LENGTH);
+
+    Mockito.when(mLoadJobConfig.getFilePath()).thenReturn(LOAD);
+    Mockito.when(mMigrateJobConfig.getSource()).thenReturn(MIGRATE);
+    Mockito.when(mPersistJobConfig.getFilePath()).thenReturn(PERSIST);
+
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(LOAD))).thenReturn(mLoadURIStatus);
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(LOAD)).getLength())
+            .thenReturn(LOAD_FILE_LENGTH);
+
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(MIGRATE))).thenReturn(mMigrateURIStatus);
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(MIGRATE)).getLength())
+            .thenReturn(MIGRATE_FILE_LENGTH);
+
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(PERSIST))).thenReturn(mPersistURIStatus);
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(PERSIST)).getLength())
+            .thenReturn(PERSIST_FILE_LENGTH);
+
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(NOOP_PLAN))).thenReturn(mNoopPlanURIStatus);
+    Mockito.when(mFileSystem.getStatus(new AlluxioURI(NOOP_PLAN)).getLength())
+            .thenReturn(NOOP_PLAN_FILE_LENGTH);
+  }
+
+  @Test
+  public void testSupportedConfigForCompleteStatus() throws Exception {
+    initialize();
+    //Run for supported configs.
+    DistributedCmdMetrics.incrementForCompleteStatusWithRetry(
```

Comment:
if you had accidentally wired the config -> metric incorrectly, such as load config increments migrate metric, this test would not be able to catch it. the test is more robust if it runs a single job -> check the corresponding metric, repeat for each job

Reply:
Agreed, let me check if it's possible to split them into several tests.

Reply:
Previously I set different sizes for different configs, see LOAD_FILE_LENGTH=1, MIGRATE_FILE_LENGTH=2 for example in the code above. So if the configs are messed up, it will still catch as the test would fail because the file size would not match metric count, for each config type.
I split the test into individual tests so it would be cleaner to see for each test.